### PR TITLE
[exporter/tanzuobservability] Support exponential histograms

### PR DESCRIPTION
**Pull request by keep94**
**Description:**
Support exponential histograms.

**Testing:**
Unit tests.

### CODE REVIEW

File: exporter/tanzuobservabilityexporter/metrics.go [hunk 1]
```diff
@@ -610,3 +614,76 @@ func (regularHistogramConsumerSpec) AsHistogram(metric pdata.Metric) histogramMe
...
 	}
 }
+
+type exponentialHistogramDataPoint struct {
+	pdata.ExponentialHistogramDataPoint
+	bucketCounts   []uint64
+	explicitBounds []float64
+}
+
+func newExponentialHistogramDataPoint(dataPoint pdata.ExponentialHistogramDataPoint) histogramDataPoint {
```
Comment (thepeterstone) [inline exporter/tanzuobservabilityexporter/metrics.go:624]: This function is pretty long and I find myself getting lost. If there's a simple way to break out some submethods to structure it a bit I think that would help keep it readable

**Review by thepeterstone — APPROVED** [review]

File: exporter/tanzuobservabilityexporter/metrics.go (hunk 1 above)
Comment (keep94) [reply exporter/tanzuobservabilityexporter/metrics.go:630]: Thank you for pointing out that this code is confusing.  I added lots of comments explaining the nuances of exponential histograms and broke the function down into several smaller ones.  Let me know if it makes more sense.

**Comment by keep94** [comment]
I am aware that load-tests are failing for my PR, but when I look at the failures, it appears that the Trace10kSPS/Zipkin test is failing.  I don't see how this failure is connected to the code that I added.  Could it be a flaky test?

File: exporter/tanzuobservabilityexporter/metrics.go (hunk 1 above)
Comment (bogdandrutu) [reply exporter/tanzuobservabilityexporter/metrics.go:695]: Very nice code :)

**Review by bogdandrutu — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
I don't believe the failure is related, I've created an issue to address the flaky test. The code looks good to me.

**Pull request merged**